internal/sigstore/verify.go

```diff
@@ -55,11 +68,8 @@ func (CosignVerifier) VerifySignature(content, signature, publicKey []byte) erro
 }
 
 // CosignPublicKeyForVersion returns the public key for the given version.
-func CosignPublicKeyForVersion(ver versionsapi.Version) ([]byte, error) {
-	if err := ver.Validate(); err != nil {
-		return nil, fmt.Errorf("selecting public key: invalid version %s: %w", ver.ShortPath(), err)
-	}
-	if ver.Ref == versionsapi.ReleaseRef && ver.Stream == "stable" {
+func CosignPublicKeyForVersion(v version) ([]byte, error) {
+	if v.Ref() == constants.ReleaseRef && v.Stream() == "stable" {
```

Comment:
If we intend to make sigstore general and not specfic to version signing, we should not have this function in this package? I don't see the benefit of having an interface if we don't have another implementation making use of it. Otherwise we decouple from the specific versionapi import, but couple through the constant and are spreading versionapi logic. Could you maybe make AzureSNP use your logic

I think that the right way to do it will emerge when we actually implement the signing logic for sth other than version.
I see that for Attestationconfigapi we are hardcoding the key in the pkg. I think we should do the same for version then.

Reply:
This function is necessary for image measurements. 
It is not required to sign version objects in the API. There are no such objects.
The interface is called `version` because image versions are specified by ref and stream.
The interface is necessary to remove the import of `internal/api/versionsapi`, so `internal/api` can import `internal/sigstore` without a cyclic import.

Do you have a location in mind that you see more fitting?

Reply:
for visibility: we agreed that there is currently no concrete need for refactoring measurement handling (as in: upcoming feature that require changes to that code). 

adding patch with subpkg.